Wikipedia/UI-V5/WMFHomeViewController.m

```diff
@@ -522,7 +522,10 @@ - (BOOL)tableView:(UITableView*)tableView shouldHighlightRowAtIndexPath:(NSIndex
 
 - (void)tableView:(UITableView*)tableView didSelectRowAtIndexPath:(NSIndexPath*)indexPath {
     id<WMFHomeSectionController> controller = [self sectionControllerForSectionAtIndex:indexPath.section];
-    MWKTitle* title                         = [controller titleForItemAtIndex:indexPath.row];
+    if ([controller respondsToSelector:@selector(shouldSelectItemAtIndex:)] && ![controller shouldSelectItemAtIndex:indexPath.item]) {
+        return;
```

Comment:
Returning `NO` from `- (BOOL)tableView:(UITableView*)tableView shouldHighlightRowAtIndexPath:(NSIndexPath*)indexPath` only disables the selection highlight. It does not prevent the selection itself. 

After doing that, you still need to ignore the selection in `- (void)tableView:(UITableView*)tableView didSelectRowAtIndexPath:(NSIndexPath*)indexPath`